### ISSUE TITLE
F821 false positive: closure captures variable before `del` statement

### ISSUE THREAD

**Issue by lamteteeow** [astral-sh/ruff#20753]
## Description

Ruff incorrectly reports F821 "Undefined name" when a variable is captured by a nested function (closure) before being deleted with `del` in the outer scope.
This has been occurring since ruff 0.13.3 first time I encountered it.

## Minimal Reproduction

```python
def main(A: int) -> int:
    A = A + 1
    def func_inner(B: int) -> int:
        return A + B  # F821: Undefined name `A` <- FALSE POSITIVE
    C = func_inner(3)
    del A
    return C

if __name__ == "__main__":
    result = main(10)
    print(result)  # Outputs: 14
```

### Version

ruff 0.14.0 (2025-10-07)

**Comment by ntBre** [comment]
Thanks for the report! I think this is a duplicate of #9858. It's a tricky bug to fix, unfortunately.

**Comment by lamteteeow** [comment]
Thanks for quick response to this. It does not bother me too much. I saw the other issue but I was not sure they rooted from the same problem so deliberately posted another one. Sorry for that.

**Comment by ntBre** [comment]
No problem at all, it's good to double check! Thanks again :)
